libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/lifecycle/UICoordinator.kt

```diff
@@ -19,13 +22,15 @@ import kotlinx.coroutines.launch
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 abstract class UICoordinator<T : ViewGroup>(
     private val viewGroup: T
-) : UIComponent() {
+) : MapboxNavigationObserver {
+
+    lateinit var coroutineScope: CoroutineScope
 
     @CallSuper
     override fun onAttached(mapboxNavigation: MapboxNavigation) {
         var attachedObserver: MapboxNavigationObserver? = null
+        coroutineScope = MainScope()
```

Comment:
For consistency I suggest using the InternalJobControlFactory to get a CoroutineScope.